dev-db/clickhouse/clickhouse-9999.ebuild

```diff
@@ -0,0 +1,724 @@
+	use cpu_flags_x86_avx2 && append-cflags "-mavx2" && append-cxxflags "-mavx2"
+	use cpu_flags_x86_popcnt && append-cflags "-mpopcnt" && append-cxxflags "-mpopcnt"
+
+	if use ppc64; then
+		 append-cflags "-maltivec" "-D__SSE2__=1" "-DNO_WARN_X86_INTRINSICS"
+		 append-cxxflags "-maltivec" "-D__SSE2__=1" "-DNO_WARN_X86_INTRINSICS"
+	fi
+
+	local line mycmakeargs use_flag cmake_option cmake_var
+	mycmakeargs=(
+		-D"CMAKE_INTERPROCEDURAL_OPTIMIZATION"="$(usex lto ON OFF)"
+		-D"DISABLE_CPU_OPTIMIZE"="ON"
+		-D"HAVE_SSE41"="$(usex cpu_flags_x86_sse4_1)"
+		-D"HAVE_SSE42"="$(usex cpu_flags_x86_sse4_2)"
+		-D"HAVE_SSSE3"="$(usex cpu_flags_x86_ssse3)"
+		-D"HAVE_AVX"="$(usex cpu_flags_x86_avx)"
+		-D"HAVE_AVX2"="$(usex cpu_flags_x86_avx2)"
+		-D"HAVE_POPCNT"="$(usex cpu_flags_x86_popcnt)"
+	)
+
+	for line in "${CLICKHOUSE_CMAKE_USE_ARGUMENTS[@]}"; do
+		read -r cmake_option use_flag <<< "${line}"
+		mycmakeargs+=( "-D${cmake_option}=$(
+			[[ ${use_flag} ]] && usex "${use_flag}" ON OFF || echo ON
+		)" )
+	done
+
+	mycmakeargs+=(
+		-D"BUILD_SHARED_LIBS"="$(usex static-libs OFF ON)"
+		-D"USE_STATIC_LIBRARIES"="$(usex static ON OFF)"
+		-D"MAKE_STATIC_LIBRARIES"="$(usex static-libs ON OFF)"
+		-D"CLICKHOUSE_SPLIT_BINARY"="$(usex split-binary ON OFF)"
+
+		-D"ENABLE_IPO"="$(usex lto ON OFF)"
+
+		-D"ENABLE_TESTS"="$(usex test ON OFF)"
+		-D"ENABLE_CODE_QUALITY"="$(usex test ON OFF)"
+		-D"WITH_COVERAGE"="$(usex test ON OFF)"
+		-D"CMAKE_DISABLE_FIND_PACKAGE_GTest"="$(usex test OFF ON)"
+
+		-D"ENABLE_CLICKHOUSE_ALL"="$(usex tools ON OFF)"
+		-D"ENABLE_CLICKHOUSE_SERVER"="$(usex server ON OFF)"
+		-D"ENABLE_CLICKHOUSE_CLIENT"="$(usex client ON OFF)"
+		-D"ENABLE_CLICKHOUSE_ODBC_BRIDGE"="$(usex odbc "$(usex tools ON OFF)" OFF)"
+		-D"ENABLE_UTILS"="$(usex tools ON OFF)"
+
+		# setting it to ON breaks configuration process because
+		# ClickHouse is not trying to set up link targets and other
+		# compile options even when this is ON
+		-D"UNBUNDLED"="OFF"
+		-D"NO_WERROR"="ON"
+
+		-D"DISABLE_INTERNAL_OPENSSL"="ON"
+		-D"ENABLE_POCO_NETSSL"="$(usex ssl ON OFF)"
+
+		-D"USE_INTERNAL_SSL_LIBRARY"="OFF"
+		-D"USE_INTERNAL_RE2_LIBRARY"="$(usex static ON OFF)"
+		-D"USE_INTERNAL_RAPIDJSON_LIBRARY"="$(usex json ON OFF)"
+		-D"USE_INTERNAL_SIMDJSON_LIBRARY"="$(usex json ON OFF)"
+
+		-D"ENABLE_RAPIDJSON"="$(usex json $(usex cpu_flags_x86_avx2 OFF ON) OFF)"
+		-D"USE_SIMDJSON"="$(usex json $(usex cpu_flags_x86_avx2 OFF ON) OFF)"
+	)
+
+	for line in "${CLICKHOUSE_CMAKE_VENDORING[@]}"; do
+		read -r cmake_option use_flag <<< "${line}"
+		[[ "${use_flag}" ]] && use "!${use_flag}" && cmake_option=${cmake_option/=ON/=OFF}
+		mycmakeargs+=( -D"${cmake_option}" )
+	done
+
+	if use llvm; then
+		mycmakeargs+=( -D"LLVM_VERSION"="$("$(get_llvm_prefix -b)"/bin/llvm-config --version)" )
+	fi
+
+	cmake-utils_src_configure
+}
+
+src_compile() {
+	cmake-utils_src_compile
+
+	if use doc; then
+		pushd "${S}/docs/tools" || die "Failed to enter docs build directory"
```

Comment:
You are inside `${S}`, so no need for absolute path here. Or for `pushd/popd` since you don't do anything else.

Reply:
on it (I just copied that from previous ebuild version where that was performed at `src_install` stage)